src/loginAzurePowerShell.ts

```diff
@@ -0,0 +1,76 @@
+import * as os from 'os';
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+import * as io from '@actions/io';
+
+var psPath: string;
+
+export const initializeAz = async (servicePrincipalId: string, servicePrincipalKey: string, tenantId: string, subscriptionId: string) => {
+    psPath = await io.which("pwsh", true);
+    await importModule();
+    await loginToAzure(servicePrincipalId, servicePrincipalKey, tenantId, subscriptionId);
+}
+
+async function importModule() {
```

Comment:
add some comments as to what this function does. Internally it calls setPSModulePath twice which is not very clear to understand why is it done.